homeassistant/components/mqtt/siren.py

```diff
@@ -335,15 +347,14 @@ async def _async_publish(
         variables: dict[str, Any] | None = None,
     ) -> None:
         """Publish MQTT payload with optional command template."""
-        template_variables = {STATE: value}
+        template_variables: dict[str, Any] = {STATE: value}
         if variables is not None:
             template_variables.update(variables)
-        payload = (
-            self._command_templates[template](value, template_variables)
-            if self._command_templates[template]
-            else json_dumps(template_variables)
-        )
-        if payload and payload not in PAYLOAD_NONE:
+        if (command_template := self._command_templates[template]) is not None:
+            payload = command_template(value, template_variables)
+        else:
+            payload = json_dumps(template_variables)
+        if payload and str(payload) not in PAYLOAD_NONE:
```

Comment:
I'm not sure what this is trying to achieve: `str(payload) not in PAYLOAD_NONE`

Should it not be?
```suggestion
        if payload and payload != PAYLOAD_NONE:
```